internals/schema_migration.py

```diff
@@ -701,3 +701,63 @@ def get_template_data(self, **kwargs) -> str:
         count += len(stages_to_update)
 
     return f'{count} subject line fields migrated.'
+
+
+class MigrateLGTMFields(FlaskHandler):
+
+  def get_template_data(self, **kwargs) -> str:
+    """Migrates old Feature subject lgtms to their respective votes."""
+    self.require_cron_header()
+
+    count = 0
+    votes_to_create = []
+    for f in Feature.query():
+      if (f.i2e_lgtms and not self.has_existing_vote(f.i2e_lgtms, GATE_API_ORIGIN_TRIAL, f)):
+        # i2e_lgtms (Intent to Experiment)'s gate_type is GATE_API_ORIGIN_TRIAL.
+        votes_to_create.append(self.create_new_vote(
+            f.i2e_lgtms, GATE_API_ORIGIN_TRIAL, f))
+        count += 1
+
+      if (f.i2s_lgtms and not self.has_existing_vote(f.i2s_lgtms, GATE_API_SHIP, f)):
+        # i2s_lgtms (Intent to Ship)'s gate_type is GATE_API_SHIP.
+        votes_to_create.append(self.create_new_vote(
+            f.i2s_lgtms, GATE_API_SHIP, f))
+        count += 1
+
+    ndb.put_multi(votes_to_create)
```

Comment:
Creating the Votes in a batch is good because it is fast.  However, when we set votes using approval_defs.set_vote() that also updates the state of the Gate.

I imagine three ways to address it:

1. Do a loop here to call set_vote() so that the Gate state is updated.  That will be much slower, but it could make partial progress on each run of the migration script so long as you don't try to write Votes for features that already have Votes for that gate_type.
2. Change set_vote() to call a new function set_votes_batch() that could set N Votes and update all the relevant Gate's at one time using put_multi() calls.
3. Allow the Votes to be written without updating the Gate states, but then run /admin/schema_migration_gate_status to update all the Gate states afterwards.  I had previously run this successfully in prod, but after you run your migration script, there will be more Votes, so this might timeout.
4. Or, we could change our approach to migration scripts so that each request to run a migration script has `start` and `num` query string parameters.  That would allow us to loop over a subset of Feature entities on each script run.  E.g., try updating 500 features at a time.

Reply:
I tweaked your idea a little bit and cap the number of votes created at 100 at a time. We could repeated hit the end point until the new vote created is 0. WDYT?